retry.go

```diff
@@ -279,14 +337,23 @@ func (r *retryHTTP) execute() (res *http.Response, err error) {
 			}
 		}
 		retryCounter++
-		if rIDReplacer == nil {
-			rIDReplacer = newRequestGUIDReplace(r.fullURL)
+		if requestGUIDReplacer == nil {
+			requestGUIDReplacer = newRequestGUIDReplace(r.fullURL)
+		}
+		r.fullURL = requestGUIDReplacer.replace()
+		if retryCountUpdater == nil {
+			retryCountUpdater = newRetryCountUpdater(r.fullURL)
+		}
+		r.fullURL = retryCountUpdater.replaceOrAdd(retryCounter)
+		if retryReasonUpdater == nil {
+			retryReasonUpdater = newRetryReasonUpdater(r.fullURL)
 		}
-		r.fullURL = rIDReplacer.replace()
-		if rUpdater == nil {
-			rUpdater = newRetryUpdate(r.fullURL)
+		retryReason := 0
+		if res != nil {
+			retryReason = res.StatusCode
 		}
-		r.fullURL = rUpdater.replaceOrAdd(retryCounter)
+		r.fullURL = retryReasonUpdater.replaceOrAdd(retryReason)
+		r.fullURL = ensureClientStartTimeIsSet(r.fullURL, clientStartTime)
 		logger.WithContext(r.ctx).Infof("sleeping %v. to timeout: %v. retrying", sleepTime, totalTimeout)
```

Comment:
Can we log the retry count and retry reason?

Reply:
Good idea, added!